pennylane/ops/op_math/linear_combination.py

```diff
@@ -341,6 +341,10 @@ def _obs_data(self):
         {(1, frozenset({('Prod', <Wires = [0, 1]>, ())})),
          (1, frozenset({('PauliZ', <Wires = [0]>, ())}))}
         """
+        warnings.warn(
+            "Accessing _obs_data is deprecated. You can still use LinearCombination.compare to check mathematical equivalence with other operators.",
+            qml.PennyLaneDeprecationWarning,
+        )
```

Comment:
@albi3ro do you think it is still necessary to do a deprecation cycle for a void private property, `_obs_data`? Or can we just go ahead and remove it all-together?

I dont really know where this is used other than `qml.Hamiltonian.compare`, and I changed that method (in `LinearCombination`) such that it is not using this private attribute anymore.

Reply:
After offline discussion decided to just remove this private property